File: leetcode/2244-minimum-rounds-to-complete-all-tasks/solution.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        """
        RUNTIME: 1027 ms (74.75%)
        MEMORY: 28.2 MB (82.79%)
        """
        hashmap = {}
        for i in tasks:
            if i not in hashmap:
                hashmap[i] = 0
            hashmap[i] += 1
        if 1 in hashmap.values(): return -1
        answer = 0
        for i in hashmap:
            if hashmap[i] == 3 or hashmap[i] == 2:
                answer += 1
            elif hashmap[i] % 3 == 0:
                answer += hashmap[i] // 3
            else:
                answer += hashmap[i] // 3 + 1
        return answer

File: leetcode/2244-minimum-rounds-to-complete-all-tasks/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 223 ms (91.63%)
     * MEMORY: 103.7 MB (45.14%)
    */
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> hashmap;
        for (int i : tasks) {
            ++hashmap[i];
        }
        int answer = 0;
        for (auto pair : hashmap) {
            if (pair.second == 1) {
                return -1;
            }
            else if (pair.second == 2 || pair.second == 3) {
                answer++;
            }
            else if (pair.second % 3 == 0) {
                answer += pair.second / 3;
            }
            else {
                answer += pair.second / 3 + 1;
            }
        }
        return answer;
    }
};